x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/processors/grok/generate_pattern_button.tsx

```diff
@@ -140,17 +141,47 @@ export const GeneratePatternButton = ({
             </EuiFlexItem>
           )}
       </EuiFlexGroup>
-      {aiFeatures.isManagedAIConnector && !aiFeatures.hasAcknowledgedAdditionalCharges && (
-        <EuiCallOut onDismiss={() => aiFeatures.acknowledgeAdditionalCharges(true)}>
-          {i18n.translate(
-            'xpack.streams.streamDetailView.managementTab.enrichment.processorFlyout.managedConnectorTooltip',
-            {
-              defaultMessage:
-                'Generating patterns is powered by a preconfigured LLM. Additional charges apply',
-            }
-          )}
-        </EuiCallOut>
-      )}
     </>
   );
 };
+
+export interface AdditionalChargesCalloutProps {
+  aiFeatures: AIFeatures;
+}
+
+export const AdditionalChargesCallout = ({ aiFeatures }: AdditionalChargesCalloutProps) => {
+  const {
+    core: { docLinks },
+  } = useKibana();
+
+  return (
+    <EuiCallOut onDismiss={() => aiFeatures.acknowledgeAdditionalCharges(true)}>
+      <FormattedMessage
+        id="xpack.streams.streamDetailView.managementTab.enrichment.processorFlyout.managedConnectorTooltip"
+        defaultMessage="Elastic LLM is the new default for generating patterns and incurs <costLink>additional charges</costLink>. Other LLM connectors remain available. <learnMoreLink>Learn more</learnMoreLink>"
+        values={{
+          costLink: (...chunks: React.ReactNode[]) => (
+            <EuiLink
+              href={docLinks?.links?.observability?.elasticManagedLlmUsageCost}
+              target="_blank"
+              rel="noopener noreferrer"
+              external
+            >
+              {chunks}
+            </EuiLink>
+          ),
+          learnMoreLink: (...chunks: React.ReactNode[]) => (
+            <EuiLink
+              href={docLinks?.links?.observability?.elasticManagedLlm}
+              target="_blank"
+              rel="noopener noreferrer"
+              external
+            >
+              {chunks}
+            </EuiLink>
+          ),
```

Comment:
These would result in broken links if what is passed to `href` is undefined due to optional chaining right? Should we do a conditional check and only render the text if those links are not retrieved?

Reply:
Agree, a dead link isn't the best experience but there's not really a great way of handling this unless you want to go down the route of writing and showing bespoke copy for when the links are defined and when they aren't. (It's not just a matter of hiding the links, the copy doesn't make sense if there aren't any). 

Considering that this is an extreme edge case and not likely to happen I would consider that overkill.